BucketeerTests/ApiClientTests.swift

```diff
@@ -826,12 +826,11 @@ class ApiClientTests: XCTestCase {
                 XCTFail()
             case .failure(let error):
                 guard
-                    let error = error as? DecodingError,
-                    case .keyNotFound(let codingKey, _) = error else {
+                    let error = error as? ResponseError,
+                    case .unacceptableCode(let code, _) = error, code == 400 else {
```

Comment:
updated: test for check the bug (could not parser error when the body is empty or is not in expected JSON format)